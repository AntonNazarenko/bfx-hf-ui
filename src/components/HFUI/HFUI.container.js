import { connect } from 'react-redux'

import UIActions from '../../redux/actions/ui'
import { getActiveMarket } from '../../redux/selectors/ui'

import HFUI from './HFUI'

const mapStateToProps = (state = {}) => ({
  activeMarket: getActiveMarket(state),
})

const mapDispatchToProps = dispatch => ({
  saveLayout: (layout, id) => {
    dispatch(UIActions.saveLayout(layout, id))
  },

  saveActiveMarket: (market) => {
    dispatch(UIActions.saveActiveMarket(market))
  },

  navigate: (route) => {
    dispatch(UIActions.setRoute(route))
  },
})

export default connect(mapStateToProps, mapDispatchToProps)(HFUI)
