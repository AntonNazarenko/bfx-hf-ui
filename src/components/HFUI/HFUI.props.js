import PropTypes from 'prop-types'

export const propTypes = {
  loginWithAuthToken: PropTypes.func.isRequired,
}

export const defaultProps = {}
