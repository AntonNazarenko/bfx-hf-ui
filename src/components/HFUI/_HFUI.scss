.hfui .hfui-api-combo-dialog-menu {
  width: 600px;
  margin: auto;
  margin-top: 100px
}

.hfui.loading {
  .bp3-spinner {
    display: block;
    margin: 100px auto 0 auto;
  }
}

.hfui, .hfui_content__wrapper {
  height: 100%;
}

.hfui_content__wrapper {
  display: flex;
  flex-direction: row;
}


.hfui_view__wrapper {
  padding: 0 32px;
  display: flex;
  flex: 1;
  flex-direction: column;

  .panel__wrapper {
    margin-bottom: 32px;

    &:last-child {
      margin-bottom: 0;
    }
  }
}
