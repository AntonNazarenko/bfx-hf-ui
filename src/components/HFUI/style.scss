.hfui-app {
  height: 100%;
}
.tos { 
  overflow: scroll;
  .hfui-modal__actions { 
    margin-top: 30px;
  }
}
.tos-content {
  overflow-x: auto; /* Use horizontal scroller if needed */
  white-space: pre-wrap; /* css-3 */
  white-space: -moz-pre-wrap !important; /* Mozilla, since 1999 */
  word-wrap: break-word; /* Internet Explorer 5.5+ */
  white-space : normal;
  max-height: 400px;
}
.hfui-app__content {
  // display: flex;
  // flex-direction: row;

  .hfui-sidebar__wrapper {
    border-right: 1px solid #333;
  }
}

.hfui-app__content-wrapper {
  display: flex;
  flex-direction: column;
  width: 100%;
}

.hfui-app__content-primary, .hfui-app__content-secondary {
  display: flex;
  flex-direction: row;
}

.hfui-app__content-primary {
  width: 100%;
  min-height: 400px;
  z-index: 1;

  .hfui-sidebar__wrapper {
    border-right: 1px solid #333;
  }

  .hfui-chart__wrapper {
    width: 100%;
  }
}
