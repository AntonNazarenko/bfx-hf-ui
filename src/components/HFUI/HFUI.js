import React from 'react'
import { Route, Switch } from 'react-router'

import TradingPage from '../../pages/Trading'
import StrategyEditorPage from '../../pages/StrategyEditor'
import MarketDataPage from '../../pages/MarketData'

import Navbar from '../Navbar'
import LoginModal from '../LoginModal'
import RegisterModal from '../RegisterModal'
import ExchangeInfoBar from '../ExchangeInfoBar'
import NotificationsSidebar from '../NotificationsSidebar'

import { propTypes, defaultProps } from './HFUI.props'
import './style.css'

export default class HFUI extends React.Component {
  static propTypes = propTypes
  static defaultProps = defaultProps

  state = {
    loginModalOpen: false,
    registerModalOpen: false,
  }

  constructor(props) {
    super(props)

    this.onChangeMarket = this.onChangeMarket.bind(this)
    this.onToggleLoginModal = this.onToggleLoginModal.bind(this)
    this.onToggleRegisterModal = this.onToggleRegisterModal.bind(this)
  }

  componentDidMount() {
    const { loginWithAuthToken } = this.props
    const cookies = document.cookie.split(';')

    cookies.forEach((c) => {
      const cookie = c.split('=')

      if (cookie[0].trim() === 'authToken') {
        loginWithAuthToken(cookie[1])
      }
    })
  }

  onChangeMarket(option) {
    const { saveActiveMarket } = this.props
    saveActiveMarket(option.value)
  }

  onToggleLoginModal() {
    this.setState(({ loginModalOpen }) => ({
      loginModalOpen: !loginModalOpen,
      registerModalOpen: false,
    }))
  }

  onToggleRegisterModal() {
    this.setState(({ registerModalOpen }) => ({
      registerModalOpen: !registerModalOpen,
      loginModalOpen: false,
    }))
  }

  render() {
    const { loginModalOpen, registerModalOpen } = this.state
    const { activeMarket } = this.props

    return (
      <div className='hfui-app'>
        <Navbar
          onLogin={this.onToggleLoginModal}
          onRegister={this.onToggleRegisterModal}
        />

        <ExchangeInfoBar
          selectedMarket={activeMarket}
          onChangeMarket={this.onChangeMarket}
        />

        <Switch>
          <Route
            exact
            path='/'
            render={() => (
              <TradingPage
                onRegister={this.onToggleRegisterModal}
                onLogin={this.onToggleLoginModal}
              />
            )}
          />

          <Route
            path='/strategy-editor'
            render={() => (
              <StrategyEditorPage onLogin={this.onToggleLoginModal} />
            )}
          />

          <Route
            path='/data'
            render={() => (
              <MarketDataPage
                onRegister={this.onToggleRegisterModal}
                onLogin={this.onToggleLoginModal}
              />
            )}
          />
        </Switch>

        <NotificationsSidebar />

        {loginModalOpen && (
          <LoginModal
            onClose={this.onToggleLoginModal}
            onRegister={this.onToggleRegisterModal}
          />
        )}

        {registerModalOpen && (
          <RegisterModal
            onClose={this.onToggleRegisterModal}
            onLogin={this.onToggleLoginModal}
          />
        )}
      </div>
    )
  }
}
