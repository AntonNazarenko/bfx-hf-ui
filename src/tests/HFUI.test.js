import React from 'react'
import renderer from 'react-test-renderer'


import HFUI from '../components/HFUI/HFUI'

// test('is HFUI always renders correct', () => {
//   const component = renderer.create(
//     // <HFUI />,
//   )
//   const tree = component.toJSON()
//   expect(tree).toMatchSnapshot()
// })
