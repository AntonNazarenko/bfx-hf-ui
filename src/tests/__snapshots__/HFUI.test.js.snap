// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`is HFUI always renders correct 1`] = `null`;
